lms/djangoapps/mobile_api/video_outlines/tests.py

```diff
@@ -410,6 +428,120 @@ class TestVideoSummaryList(
     """
     REVERSE_INFO = {'name': 'video-summary-list', 'params': ['course_id']}
 
+    def test_mobile_api_config(self):
+        """
+        Tests VideoSummaryList with different MobileApiConfig video_profiles
+        """
+        self.login_and_enroll()
+        edx_video_id = "testing_mobile_high"
+        api.create_video({
+            'edx_video_id': edx_video_id,
+            'status': 'test',
+            'client_video_id': u"test video omega \u03a9",
+            'duration': 12,
+            'courses': [unicode(self.course.id)],
+            'encoded_videos': [
+                {
+                    'profile': 'youtube',
+                    'url': 'xyz123',
+                    'file_size': 2222,
+                    'bitrate': 4444
+                },
+                {
+                    'profile': 'mobile_high',
+                    'url': self.video_url_high,
+                    'file_size': 111,
+                    'bitrate': 333
+                },
+
+            ]})
+        ItemFactory.create(
+            parent=self.other_unit,
+            category="video",
+            display_name=u"testing mobile high video",
+            edx_video_id=edx_video_id,
+        )
+        course_outline = self.api_response().data
+
+        expected_output = {
+            'category': u'video',
+            'video_thumbnail_url': None,
+            'language': u'en',
+            'name': u'testing mobile high video',
+            'video_url': self.video_url_high,
+            'duration': 12.0,
+            'transcripts': {
+                'en': 'http://testserver/api/mobile/v0.5/video_outlines/transcripts/{}/testing_mobile_high_video/en'.format(self.course.id)  # pylint: disable=line-too-long
+            },
+            'encoded_videos': {
+                u'mobile_high': {
+                    'url': self.video_url_high,
+                    'file_size': 111
+                },
+                u'youtube': {
+                    'url': u'xyz123',
+                    'file_size': 2222
+                }
+            },
+            'size': 111
+        }
+        course_outline[0]['summary'].pop("id")
+        self.assertEqual(course_outline[0]['summary'], expected_output)
+
+        MobileApiConfig(video_profiles="mobile_low,youtube").save()
```

Comment:
This test case has gotten pretty long. There are a couple of things that you can do to make it shorter and clearer.

For each MobileApiConfig change, add a comment about what the change really means (i.e. "Include different profiles" or "Change the priority order of the profiles").

Instead of making a whole new expected_output dict, just change the relevant keys. That way it is clear exactly what effect the MobileApiConfig change has on the output. Also, with this change, it will be better if the order swapping case comes before the changing profiles case.